torch/testing/_internal/common_utils.py

```diff
@@ -1987,22 +1989,26 @@ def make_tensor(size, device: torch.device, dtype: torch.dtype, *, low=None, hig
         complex type and 1 if integer/boolean type.
     """
 
-    assert low is None or low < 9, "low value too high!"
-    assert high is None or high > -9, "high value too low!"
+    if low is not None and high is not None:
+        assert low <= high, "Expected low value to be lower than or equal to high, but found higher instead."
+        assert high >= low, "Expected high value to be higher than or equal to low, but found lower instead."
 
     if dtype is torch.bool:
         result = torch.randint(0, 2, size, device=device, dtype=dtype)
     elif dtype is torch.uint8:
-        low = math.floor(0 if low is None else max(low, 0))
-        high = math.ceil(10 if high is None else min(high, 10))
+        ranges = (torch.iinfo(dtype).min, torch.iinfo(dtype).max)
+        low = math.floor(ranges[0] if low is float('-inf') else 0 if low is None or low < ranges[0] else low)
```

Comment:
This silently clamps the lower bound to `0` in case the user passes a negative value. Shouldn't we simply raise an error, since the function is clearly used wrong?

Reply:
Good point, if we think of this going as a public facing API, raising an error sounds a good idea. Let's raise an error if `low < ranges[0]` (and low is not `float('-inf')`) or if `high > ranges[1]` for all the dtypes. What do you think?

Reply:
The same of course applies to `high > ranges[1]` and also to all other branches.

Reply:
Okay, so we might to re-think on this point. Here is why:

`make_tensor` is internally used as: (example usage: https://github.com/pytorch/pytorch/blob/master/test/test_binary_ufuncs.py#L1674)

```
def some_function(device, some_dtype):
    make_tensor(size, device=device, dtype=some_dtype, low=-9, high=9)
```

low, high as `(-9, 9)` works for integral types like int, int64 but not for `uint8`. In these cases, the usage expects `make_tensor` to clamp the low and high values depending on the dtype.

The same is possible for public facing API, when they want to pass a generic (low, high) value but expect `make_tensor` to clamp for `uint8`.

```
# Example use case when you are making sure that integral types, floating types and complex types use [-1, 1) as the domain but uint8 takes [0, 1)
make_tensor(size, device, dtype, low=-1, high=1)
```

Reply:
I see the use case, but I'm still against doing this silently and by default. As a reviewer if I read `make_tensor(..., low=-9, high=9)` I'm expecting the values to be in that range.

We could add a `clamp_to_dtype_range: bool = False` kwarg that enables this functionality. This should make it clear for the reviewer that some clamping is going on.